devlib/platform/arm.py

```diff
@@ -88,6 +88,8 @@ def init_target_connection(self, target):
     def _init_android_target(self, target):
         if target.connection_settings.get('device') is None:
             addr = self._get_target_ip_address(target)
+            # Convert bytes to string for Python3 compatibility
+            addr = addr.decode("utf-8")
```

Comment:
This needs to be guarded by `if sys.version_info[0] == 3:`

Reply:
Done